ciao-image/datastore/store.go

```diff
@@ -178,11 +178,14 @@ func (s *ImageStore) UploadImage(ID string, body io.Reader) error {
 	if s.rawDs != nil {
 		err = s.rawDs.Write(ID, body)
 		if err != nil {
+			database.Logger.Errorf("Could not write image: %v", err)
+			// TODO shouldn't we return err here?
```

Comment:
@dorileo well spotted but actually, I think it's the block of code below that should not have a return.  By returning it means the the image is not persisted.  The image will exist with state set to killed but then if you reboot the controller I think the image will dissapear, and the rbd image will be leaked.